packages/react/src/components/TextField/TextField.types.ts

```diff
@@ -75,6 +75,12 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   autoAdjustHeight?: boolean;
 
+  /**
+   * Callback executed after a multiline TextField's height is adjusted.
+   * This callback is only executed when `autoAdjustHeight` is true.
+   */
+  onAdjustHeight?: (scrollHeight: number) => void;
```

Comment:
This seems like an odd API to solve this issue. Are there other reasons besides this bug that a user would need to know when the height was adjusted?

Another option would be to have a prop like `scrollContainerRef?: React.RefObject<HTMLElement>` that lets the user pass in a ref to the element with the scrollbars. Then if it's set, the textbox will maintain the scroll position when changing the height.

```diff
private _adjustInputHeight(): void {
    if (this._textElement.current && this.props.autoAdjustHeight && this.props.multiline) {
+     const scrollContainer = this.props.scrollContainerRef?.current;
+     const scrollTop = scrollContainer?.scrollTop;
      const textField = this._textElement.current;
      textField.style.height = '';
      textField.style.height = textField.scrollHeight + 'px';
+     if (scrollContainer && scrollTop) {
+       scrollContainer.scrollTop = scrollTop;
+     }
    }
  }
```

It's also much easier for the user to get this fix: they just provide one ref instead of having to keep the scrollTop in a state variable and update it whenever the value changes.

Reply:
I kept this API because of the [comment](https://github.com/microsoft/fluentui/issues/16653#issuecomment-847469198), but I am open to just using the ref instead.

Reply:
Ah I didn't realize that option had already been discussed. The ref seems a little cleaner IMO, and it requires opting in; it's only messing with the scrollTop property if the user sets the prop. But either is fine.